server/cluster/coordinator.go

```diff
@@ -966,3 +967,17 @@ func (c *coordinator) getPausedSchedulerDelayUntil(name string) (int64, error) {
 	}
 	return s.GetDelayUntil(), nil
 }
+
+func (c *coordinator) CheckTransferWitnessLeader(region *core.RegionInfo) {
+	c.RLock()
+	defer c.RUnlock()
+	if s, ok := c.schedulers[schedulers.TransferWitnessLeaderName]; ok {
+		if schedulers.NeedTransferWitnessLeader(region) {
+			select {
```

Comment:
Will this block for a long time? I'm afraid it may slow down the heartbeat process speed. Also, what will happen if the region number is enormous? Will this become a bottleneck?

Reply:
For failover scenarios only, and the scene is filtered before sending to the channel. Therefore, there will not be a large number of regions. Even if the size of the channel is exceeded, it will be directly discarded without blocking.